### PR TITLE
Safeguard chrome using managed configuration APIs

### PR DESCRIPTION
**Pull request by ankitgupta29**

### CODE REVIEW

File: managed-configuration-chrome.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                "IncognitoModeAvailability": "1",   # disable incognito mode
+                "ForceGoogleSafeSearch" : "true",   # enable safe search to work
+                "HomepageLocation": "https://esper.io"  # set chrome home page
+            }
+        }
+    }
+)
+
+# int | Number of results to return per page. (optional) (default to 20)
+configuration.per_page_limit = 5000
```
Comment (abdulMuneerM) [inline managed-configuration-chrome.py]: Pls use a small number like 20.

File: managed-configuration-chrome.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+##  * all: Run commands on all the devices. Commands will be queued for offline devices until they come back online.
+
+# contains the group id of all the groups in the enterprise
+def run_managed_configuration():
+
+    request = esperclient.V0CommandRequest(
+        enterprise=enterprise_id,
+        command_type="GROUP",
+        device_type="all",
+        groups = allgroups,
```
Comment (abdulMuneerM) [inline managed-configuration-chrome.py]: all groups in a single command request will cause heavy load in the system. Pls try to create separate command requests for 1-3 groups.

File: managed-configuration-chrome.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    except ApiException as e:
+        print("Exception when calling CommandsV2Api->create_command: %s\n" % e)
+
+
+"""
+Main Function
+"""
+def main():
+    get_all_groups_in_enterprise()
```
Comment (tarun91vas) [inline managed-configuration-chrome.py]: do we want to run this for all groups/all devices ?

File: managed-configuration-chrome.py (hunk 1 above)
Comment (tarun91vas) [reply managed-configuration-chrome.py]: (nit)yes, better to use pagination to avoid latency.

File: managed-configuration-chrome.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        api_response = api_instance.create_command(enterprise_id, request)
+        #print(api_response)
+
+        request_id = api_response.id
+        response = api_instance.get_command_request_status(enterprise_id, request_id)
+        status = response.results[0]
+        #print(status)
+
+        while status.state not in ["Command Success", "Command Failure", "Command TimeOut", "Command Cancelled", "Command Queued"]:
+            response = api_instance.get_command_request_status(enterprise_id, request_id)
```
Comment (tarun91vas) [inline managed-configuration-chrome.py:98]: Please add some sleep, this is running in while

File: managed-configuration-chrome.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    )
+
+    try:
+        # Create a command request
+        api_response = api_instance.create_command(enterprise_id, request)
+        #print(api_response)
+
+        request_id = api_response.id
+        response = api_instance.get_command_request_status(enterprise_id, request_id)
+        status = response.results[0]
```
Comment (tarun91vas) [inline managed-configuration-chrome.py:94]: Why only first result ?

File: managed-configuration-chrome.py (hunk 2 above)
Comment (ankitgupta29) [reply managed-configuration-chrome.py]: I have moved run_managed_configuration inside groups calculation and calling for each times

File: managed-configuration-chrome.py (hunk 3 above)
Comment (ankitgupta29) [reply managed-configuration-chrome.py]: Does it matter? Its an example only and people can replace it with devices if they want

File: managed-configuration-chrome.py (hunk 4 above)
Comment (ankitgupta29) [reply managed-configuration-chrome.py:98]: done

File: managed-configuration-chrome.py (hunk 5 above)
Comment (ankitgupta29) [reply managed-configuration-chrome.py:94]: Only latest result matters right? not the old status of the same command

File: managed-configuration-chrome.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    api_instance = esperclient.DeviceGroupApi(esperclient.ApiClient(configuration))
+    try:
+        api_response = api_instance.get_all_groups(enterprise_id,
+                                                    limit=configuration.per_page_limit,
+                                                    offset=configuration.per_page_offset)
+        if len(api_response.results):
+            for group in api_response.results:
+                # add all the devices in this group to global list of devices
+                allgroups.append(group.id)
+                run_managed_configuration(allgroups)
```
Comment (abdulMuneerM) [inline managed-configuration-chrome.py]: Here the value of 'allgroups' is like this, [1], [1,2], [1,2,3]...etc.
You have to use singleton list to 'run_managed_configuration' function if you're looking to create command request for each group.
run_managed_configuration([group.id])

File: managed-configuration-chrome.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    except ApiException as e:
+        print("Exception when calling CommandsV2Api->create_command: %s\n" % e)
+
+
+"""
+Main Function
+"""
+def main():
+    get_all_groups_in_enterprise()
+    run_managed_configuration()
```
Comment (abdulMuneerM) [inline managed-configuration-chrome.py]: Function prototype has modified. Also you're doing all operations inside 'get_all_groups_in_enterprise' function, so this function call is not needed I guess.

Comment (ankitgupta29) [reply managed-configuration-chrome.py]: Updated. Thanks

File: managed-configuration-chrome.py (hunk 6 above)
Comment (ankitgupta29) [reply managed-configuration-chrome.py]: done.

**Review by abdulMuneerM — APPROVED** [review]

File: managed-configuration-chrome.py (hunk 5 above)
Comment (tarun91vas) [reply managed-configuration-chrome.py:94]: command status is list of command on all the device for a particular request. each result will have it own status. For example you created a request for a group with 10 devices. Above call will results 10 objects in result each representing status of command for each device. This is what is different in new framework.

**Pull request merged**